File: backend/app/domain/categories.py
"""Controlled vocabulary of solution categories.

Single source of truth. The SQL CHECK constraint in 0001_schema.sql mirrors this list —
keep them in sync.
"""

from enum import StrEnum


class Category(StrEnum):
    TRANSPORT = "transport"
    ENERGY = "energy"
    HOUSING = "housing"
    WATER = "water"
    WASTE = "waste"
    SAFETY = "safety"
    HEALTH = "health"
    ENVIRONMENT = "environment"
    DIGITAL = "digital"
    GOVERNANCE = "governance"
    ECONOMY = "economy"
    CLIMATE_RESILIENCE = "climate_resilience"


CATEGORIES: list[str] = [c.value for c in Category]

# Human-friendly labels for the frontend filter UI.
CATEGORY_LABELS: dict[str, str] = {
    Category.TRANSPORT: "Транспорт і мобільність",
    Category.ENERGY: "Енергетика й опалення",
    Category.HOUSING: "Житло",
    Category.WATER: "Вода й водовідведення",
    Category.WASTE: "Поводження з відходами",
    Category.SAFETY: "Громадська безпека",
    Category.HEALTH: "Здоров'я",
    Category.ENVIRONMENT: "Довкілля й зелені зони",
    Category.DIGITAL: "Цифрове й розумне місто",
    Category.GOVERNANCE: "Врядування й участь",
    Category.ECONOMY: "Місцева економіка",
    Category.CLIMATE_RESILIENCE: "Кліматична стійкість",
}


def is_valid(category: str) -> bool:
    return category in CATEGORIES

File: frontend/src/lib/categories.ts
import type { Category } from '../api/types'

/** Static fallback for /api/v1/categories — same vocabulary, offline-safe. */
export const CATEGORY_LABELS: Record<Category, string> = {
  transport: 'Транспорт і мобільність',
  energy: 'Енергетика й опалення',
  housing: 'Житло',
  water: 'Вода й водовідведення',
  waste: 'Поводження з відходами',
  safety: 'Громадська безпека',
  health: "Здоров'я",
  environment: 'Довкілля й зелені зони',
  digital: 'Цифрове й розумне місто',
  governance: 'Врядування й участь',
  economy: 'Місцева економіка',
  climate_resilience: 'Кліматична стійкість',
}

export function categoryLabel(value: string): string {
  return CATEGORY_LABELS[value as Category] ?? value
}
